source/common/network/io_socket_handle_impl.cc

```
@@ -277,7 +277,7 @@ Api::IoCallUint64Result IoSocketHandleImpl::sendmsg(const Buffer::RawSlice* slic
 
 Address::InstanceConstSharedPtr getAddressFromSockAddrOrDie(const sockaddr_storage& ss,
                                                             socklen_t ss_len, os_fd_t fd) {
-  try {
+  TRY_NEEDS_AUDIT {
```

If we are going to die anyway, why not just let the exception propagate? Leave a comment to follow up?

---

That's a good point, I actually don't know why it should die in this case.

---

TODO added.

---

The issue is someone might possibly catch an Envoy exception. This should just be converted to not using exceptions at all and then panic, so please update the TODO

---

Thanks, will fix.

---

Looks like the TODO is updated; resolving.